packages/nep141-erc20/src/utils/index.js

```diff
@@ -0,0 +1,84 @@
+import Web3 from 'web3'
+import { getEthProvider } from '@near-eth/client/dist/utils'
+
+/**
+ * Search for a dropped and replaced transaction (speed up or cancel)
+ * @param {Number} safeHeightBeforeEthTx Lower search bound
+ * @param {*} txInfo Transaction information to find the replacement
+ * @param {*} tx.nonce Nonce of the transaction searched
+ * @param {*} tx.from Signer of the transaction searched
+ * @param {*} tx.to Address called by the transaction searched
+ * @param {*} event.name Name of event expected if tx was speed up
+ * @param {*} event.abi tx.to contract abi needed to find event
+ * @param {*} event.validate Function to validate the content of event
+ */
+export async function findReplacementTx (safeHeightBeforeEthTx, tx, event) {
+  const web3 = new Web3(getEthProvider())
+
+  const currentNonce = await web3.eth.getTransactionCount(tx.from, 'latest')
+
+  // Transaction still pending
+  if (currentNonce <= tx.nonce) return null
```

Comment:
Do I understand correctly that if the user has made _any_ Ethereum transaction since they started their transfer, we assume it is a replace/speedup? If I start a transfer, then go make a swap on Uniswap, won't this function throw an error with "Transaction was dropped and replaced by [uniswap tx hash]"?

Reply:
No because the uniswap tx will be using the next nonce and have no impact on the bridge transfer transaction

Reply:
`currentNonce` will increase due to the swap, right? And then `currentNonce > tx.nonce`, so it will proceed past this `if` and try to find the replacement.

Reply:
Correct, but the call to `getTransactionReceipt` before would most likely have returned a receipt and `findReplacementTx` is never called. If the transaction gets mined in that split second, then `findReplacementTx` will just find the original tx.